addons/stock_account/migrations/13.0.1.1/post-migration.py

```diff
@@ -253,10 +257,63 @@ def generate_stock_valuation_layer(env):
+                        if mto_elem["move_dest_id"] == move["move_orig"]:
+                            svl_in_vals_mto_filtered_list.append(mto_elem)
+                    qty = move["product_qty"]
+                    if product.cost_method in ("average", "fifo") and not is_dropship:
+                        # Reduce remaining qty in svl of type "in"
+                        while qty > 0 and svl_in_index_mto < len(svl_in_vals_mto_filtered_list):
+                            if svl_in_vals_mto_filtered_list[svl_in_index_mto]["remaining_qty"] >= qty:
+                                candidate_cost = (svl_in_vals_mto_filtered_list[svl_in_index_mto]["remaining_value"] /
+                                                  svl_in_vals_mto_filtered_list[svl_in_index_mto]["remaining_qty"])
+                                if usage:
+                                    # Prepare layer usage
+                                    svl_usage_vals = _prepare_svl_usage_vals(
+                                        svl_in_vals_mto_filtered_list[svl_in_index_mto]["id"], move, qty,
+                                        candidate_cost*qty)
+                                    svl_usage_vals_list.append(svl_usage_vals)
+                                svl_in_vals_mto_filtered_list[svl_in_index_mto]["remaining_qty"] -= qty
+                                svl_in_vals_mto_filtered_list[svl_in_index_mto]["remaining_value"] = float_round(
+                                    candidate_cost * svl_in_vals_mto_filtered_list[svl_in_index_mto]["remaining_qty"],
+                                    precision_digits=precision_price)
+                                qty = 0
+                            elif svl_in_vals_mto_filtered_list[svl_in_index_mto]["remaining_qty"]:
+                                if usage:
+                                    # Prepare layer usage
+                                    svl_usage_vals = _prepare_svl_usage_vals(
+                                        svl_in_vals_mto_filtered_list[svl_in_index_mto]["id"],
+                                        move, svl_in_vals_mto_filtered_list[svl_in_index_mto]["remaining_qty"],
+                                        svl_in_vals_mto_filtered_list[svl_in_index_mto]["remaining_value"])
+                                    svl_in_vals_mto_filtered_list.append(svl_usage_vals)
+                                qty -= svl_in_vals_mto_filtered_list[svl_in_index_mto]["remaining_qty"]
+                                svl_in_vals_mto_filtered_list[svl_in_index_mto]["remaining_qty"] = 0.0
+                                svl_in_vals_mto_filtered_list[svl_in_index_mto]["remaining_value"] = 0.0
+                                svl_in_index_mto += 1
+                            else:
+                                svl_in_vals_mto_filtered_list[svl_in_index_mto]["remaining_qty"] = 0.0
+                                svl_in_vals_mto_filtered_list[svl_in_index_mto]["remaining_value"] = 0.0
```

Comment:
```suggestion
```
is needed this or we could delete it? same question for lines 347 and 348

Reply:
I think we can delete it but I left that for security

Reply:
I will delete them anyway

Reply:
xD